workerpool_test.go

```diff
@@ -351,6 +351,34 @@ func TestStopRace(t *testing.T) {
 	close(releaseChan)
 }
 
+func TestWaitingQueueSizeRace(t *testing.T) {
+	const maxQueueSize = 1
+	wp := New(1)
+	workRelChan := make(chan struct{})
+	errChan := make(chan bool, 3)
+
+	go func() {
+		for i := 0; i < 3; i++ {
+			// Submit a new task only if # waiting in the queue is at maxQueueSize or more
+			if wp.WaitingQueueSize() > maxQueueSize-1 {
+				errChan <- true
+				continue
+			}
+			wp.Submit(func() { <-workRelChan })
```

Comment:
Need small change following this line:
```
			// Wait for task to be removed from task queue.
			for len(wp.taskQueue) != 0 {
				time.Sleep(time.Microsecond)
			}
```
Otherwise, task may still be on `taskQueue` and not yet on `waitingQueue`